### PR TITLE
Add a tag variant with rounded corners

### PR DESCRIPTION
**Pull request by tuomashatakka**
## Description

Add a tag variant with rounded corners

## Related Issue

[HDS-1327](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1327)

## How Has This Been Tested?

- On a local machine
- Automated tests

## Screenshots (if appropriate):

Screenshot 2022-11-16 at 13 49 29

### CODE REVIEW

**Review by minevala — COMMENTED** [review]
Nice work 👍 
I think the styles should be part of Core lib. And we also need to add an example of rounded tags into documentation site

**Comment by minevala** [comment]
We need to provide an example in HDS tag documentation: https://hds.hel.fi/components/tag/code

**Comment by tuomashatakka** [comment]
That's a good point with including the styles in the core package. I'll move the style definitions there.

I'll update the documentation as well.

**Comment by hannu-kivimaki** [comment]
When documenting the new variants, I noticed the large rounded tag has `--font-body-size-s`. The large variant would need  `--font-body-size-m`, which is 16px. It makes the large variant more accessible and perceivable.

Screenshot 2022-11-21 at 11 06 51

File: packages/react/src/components/tag/Tag.module.scss [hunk 1]
```diff
@@ -27,13 +28,36 @@
...
+
+.tag.rounded-corners > .label:not(:only-child) {
+  padding-right: 0;
+  padding-left:  var(--tag-padding);
+}
+
+.tag.rounded-corners > button {
+  padding-right: calc(var(--tag-padding) / 2);
+}
+
```
[the remark this answers is not in the document]
Comment (tuomashatakka) [reply packages/react/src/components/tag/Tag.module.scss]: Yes it would :)

**Review by minevala — COMMENTED** [review]
Nice work 🏅 There are a couple of Grammarly fixes in the documentation text and it is good to go.
Do we have a ticket for upgrading the tag in the next major version? The design looks a bit different than the implementation. The delete button is on the other side in sharp-corned tags, there is not as much space between the label and the delete button, and maybe the larger tag could be a bit higher.

**Review by minevala — APPROVED** [review]
Good job 🥇

**Pull request merged**